Add Backup / Restore docs (#2072)
codenvy/che-docs: modules/administration-guide/partials/con_setup-backup-server.adoc

> [id="con_setup-backup-server"]

Reviewer: ```suggestion
[id="setting-up-a-backup-server"]
```

= Setting up a backup server

> This section describes which backup servers supported in {prod-short} and gives hints on how to setup one.

Reviewer: ```suggestion
This section describes the backup servers supported in {prod-short} and provides instructions on how to set up one.
```

include::partial$snip_internal-backup-server-warning.adoc[]

> {prod-short} uses link:https://restic.readthedocs.io/en/latest/[`restic`] tool to manage backup snapshots and to push / pull backup data to / from a backup server.

Reviewer: ```suggestion
{prod-short} uses link:https://restic.readthedocs.io/en/latest/[`restic`] tool to manage backup snapshots and to push and pull backup data to and from a backup server.
```

[NOTE]
====

> `restic` is licensed under `BSD 2-Clause License`

Reviewer: ```suggestion
`restic` is licensed under the BSD 2-Clause License.
```
Is this information necessary here?

====

> The following types of backup servers are supported in {prod-short} at the moment:

Reviewer: ```suggestion
The following types of backup servers are supported in {prod-short}:
```

> * `REST`. Developed specifically for `restic`. See link:https://restic.readthedocs.io/en/latest/[how to set up REST server] documentation.

Reviewer: ```suggestion
* *REST*. Developed specifically for `restic`. See link:https://restic.readthedocs.io/en/latest/[how to set up REST server] documentation.
```

> * `Amazon S3` and API compatible alternatives. See link:https://docs.aws.amazon.com/s3/index.html[AWS S3] or your preferred alternative documentation.

Reviewer: ```suggestion
* *Amazon S3* and API-compatible alternatives. See link:https://docs.aws.amazon.com/s3/index.html[AWS S3] or your preferred alternative documentation.
```

* `SFTP`

> ifeval::["{project-context}" == "che"]

Reviewer 2: Why only for `che` ?

Author: Fixed. Added reference to RHEL docs

> * `SFTP` See link:https://access.redhat.com/solutions/2399571[How to configure SFTP server]

Reviewer: ```suggestion
* *SFTP*. See link:https://access.redhat.com/solutions/2399571[How to configure SFTP server].
```

[id="con_setup-backup-server"]
= Setting up a backup server

The following section describes the supported {prod-short} backup servers and provides information for their setup.

include::partial$snip_internal-backup-server-warning.adoc[]

{prod-short} uses the link:https://restic.readthedocs.io/en/latest/[`restic`] tool to: 

* manage backup snapshots

* push to or to pull backup data from a backup server
+
[NOTE]
====

> The `restic` backup tool is licensed under the link:https://opensource.org/licenses/BSD-2-Clause[BSD 2-Clause] license.

Reviewer: Is this piece of info necessary?

Author: Yes, it is... The reason is described [here](https://github.com/eclipse/che/issues/20028).

====

.The backup servers currently supported for {prod-short}:

> * `REST`

Reviewer: ```suggestion
REST:: The REST server is a solution designed to cooperate with the `restic` tool. See link:https://restic.readthedocs.io/en/latest/[How to set up a REST server] documentation.
```

> +

Reviewer: ```suggestion
```

> ** The `REST` server was a solution designed to co-operate with the `restic` tool.

Reviewer: ```suggestion
```

> ** See link:https://restic.readthedocs.io/en/latest/[How to set up a REST server] documentation.

Reviewer: ```suggestion
```

> * `Amazon S3` and API compatible alternatives.

Reviewer: ```suggestion
Amazon S3 and API compatible alternatives:: See link:https://docs.aws.amazon.com/s3/index.html[AWS S3 Simple Storage Service Documentation] or the docs of alternative services that have compatible API with AWS.
```

> ** See link:https://docs.aws.amazon.com/s3/index.html[AWS S3 Simple Storage Service Documentation] or the docs of alternative services that have compatible API with AWS.

Reviewer: ```suggestion
```

> * `SFTP`

Reviewer: ```suggestion
SFTP:: See link:https://access.redhat.com/solutions/2399571[How to configure an SFTP server].
```

> ** See link:https://access.redhat.com/solutions/2399571[How to configure an SFTP server]

Reviewer: ```suggestion
```
